src/nvim/autocmd_defs.h

```diff
@@ -37,10 +37,11 @@ typedef struct {
 } AutoPat;
 
 typedef struct {
-  AucmdExecutable exec;     ///< Command or callback function
   AutoPat *pat;             ///< Pattern reference (NULL when autocmd was removed)
   int64_t id;               ///< ID used for uniquely tracking an autocmd
   char *desc;               ///< Description for the autocmd
+  char *handler_cmd;        ///< Handler Ex command (NULL if handler is a function).
+  Callback handler_fn;      ///< Handler callback (ignored if `handler_cmd` is not NULL).
```

Comment:
`AucmdExecutable` adds 2 layers of indirection. It can be avoided by storing the cmd/fn fields directly on the `AutoCmd` struct.

Although formalizing a `union` is good practice for heavily-used interfaces, this struct is mainly for `autocmd_register` which is internal to this module.